animation_nodes/nodes/geometry/c_utils.pyx

```diff
@@ -0,0 +1,494 @@
+
+def intersectLineSphereList(Py_ssize_t amount,
+                            VirtualVector3DList lineStartList,
+                            VirtualVector3DList lineEndList,
+                            VirtualVector3DList sphereCenterList,
+                            VirtualDoubleList sphereRadiusList):
+    cdef Vector3DList firstIntersectionList = Vector3DList(length = amount)
+    cdef Vector3DList secondIntersectionList = Vector3DList(length = amount)
+    cdef DoubleList firstParameterList = DoubleList(length = amount)
+    cdef DoubleList secondParameterList = DoubleList(length = amount)
+    cdef CharList numberOfIntersectionsList = CharList(length = amount)
+    cdef Vector3 firstIntersection, secondIntersection
+    cdef double firstParameter, secondParameter
+    cdef char numberOfIntersections
+    cdef Py_ssize_t i
+    for i in range(amount):
+        intersectLineSphere(lineStartList.get(i), lineEndList.get(i),
+        sphereCenterList.get(i), sphereRadiusList.get(i),
+        &firstIntersection, &secondIntersection,
+        &firstParameter, &secondParameter, &numberOfIntersections)
+        firstIntersectionList.data[i] = firstIntersection
+        secondIntersectionList.data[i] = secondIntersection
+        firstParameterList.data[i] = firstParameter
+        secondParameterList.data[i] = secondParameter
+        numberOfIntersectionsList.data[i] = numberOfIntersections
+    return (firstIntersectionList, secondIntersectionList,
+            firstParameterList, secondParameterList,
+            numberOfIntersectionsList)
+
+def intersectLineSphereSingle(lineStart,
+                            lineEnd,
+                            sphereCenter,
+                            sphereRadius):
+    cdef Vector3 _lineStart = toVector3(lineStart)
+    cdef Vector3 _lineEnd = toVector3(lineEnd)
+    cdef Vector3 _sphereCenter = toVector3(sphereCenter)
+    cdef Vector3 firstIntersection, secondIntersection
+    cdef double firstParameter, secondParameter
+    cdef char numberOfIntersections
+    intersectLineSphere(&_lineStart, &_lineEnd, &_sphereCenter, sphereRadius,
+    &firstIntersection, &secondIntersection,
+    &firstParameter,&secondParameter, &numberOfIntersections)
+    return (toPyVector3(&firstIntersection), toPyVector3(&secondIntersection),
+            firstParameter, secondParameter,
+            numberOfIntersections)
+
+cdef intersectPlanePlane(Vector3 *firstPlanePoint, Vector3 *firstPlaneNormal,
+                        Vector3 *secondPlanePoint, Vector3 *secondPlaneNormal,
+                        Vector3 *outLineDirection, Vector3 *outLinePoint, bint *outValid):
+    cdef Vector3 direction, unitNormal1, unitNormal2, factor1, factor2, point
+    cdef double h1, h2, dot, c1, c2, invDotSquared
+    normalizeVec3(&unitNormal1, firstPlaneNormal)
+    normalizeVec3(&unitNormal2, secondPlaneNormal)
+    crossVec3(&direction, firstPlaneNormal, secondPlaneNormal)
+    if lengthVec3(&direction) > 1e-6:
```

Comment:
please invert the branch cases.
```
if lengthVec3(&direction) < 1e-6:
    ...
    return

...
```

This is called a guard condition: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
It helps to keep the main execution code at first level of indentation, making it easier to read.